protocol.md

```diff
@@ -726,6 +727,120 @@ Upload-Length: 11
 Upload-Concat: final;/files/a /files/b
 ```
 
+### Client Tag
+
+With this extension, Clients can provide a Tag for an upload - with which they
+can later retrieve the URL for the created resource.
+
+This allows Clients to resume an upload initiated through the [Creation With Upload](#creation-with-upload)
+or [Creation](#creation) extensions for which they did not receive the response with
+the newly created resource's URL (for example due to a broken connection).
+
+If the Server supports this extension, it MUST add `client-tag` to the `Tus-Extension`
+header.
+
+To initiate an upload with Client Tag, Clients MUST include an `Upload-Tag`
+header with the initial `POST` request (as specified in the [Creation](#creation) and
+[Creation With Upload](#creation-with-upload) extensions) made to the Upload URL.
+
+Clients can then send a `HEAD` request with the same `Upload-Tag` header to the
+Upload URL, which - upon success - MUST respond with the `200 OK` or `204 No Content`
+status, and the resource's URL in the response's `Location` header.
+
+A successful response MUST also contain the `Tus-Version` header. An `Upload-Offset` header
+that indicates how many bytes have already been received by the server MAY also be returned.
```

Comment:
Is there a situation where a server doesn't want to include the Upload-Offset header? This would save the client an additional HEAD request if we can rely on the offset being present in the response.

Reply:
Could we make it easier on ourselves and just refer to this section in addition to specifying the `Location` header? In my world these are the same bar the extra location header. https://tus.io/protocols/resumable-upload.html#head

Reply:
I like that too - and address it with the next commit.